frontend/src/lib/components/TestJobLoader.svelte

```diff
@@ -46,6 +46,9 @@
 			}
 			return testId
 		} catch (err) {
+			// if error happens on submitting the job, reset UI state so the user can try again
+			isLoading = false
+			job = undefined
```

Comment:
```suggestion
			currentId = undefined
			job = undefined
```

currentId being undefined will trigger the condition at line 26 and handle more cases

Reply:
Tried testing with this approach, but the component stays stuck in a state where it thinks it is still running.  I'll add the reset of the `currentId`, but without some other changes to the component I think we need to update `isLoading`